Add information about Allow Destroy Plan setting (#1187)
hashicorp/terraform-website: content/source/docs/cloud/api/workspaces.html.md

----------------------------------------------|---------|-----------|------------
`data.type`                                   | string  |           | Must be `"workspaces"`.
`data.attributes.name`                        | string  |           | The name of the workspace, which can only include letters, numbers, `-`, and `_`. This will be used as an identifier and must be unique in the organization.

> `data.attributes.allow-destroy-plan`          | boolean | `true`    | Whether destroy plans can be applied to the workspace.

Reviewer: ```suggestion
`data.attributes.allow-destroy-plan`          | boolean | `true`    | Whether destroy plans can be queued on the workspace.
```

`data.attributes.name`                        | string         | (previous value) | A new name for the workspace, which can only include letters, numbers, `-`, and `_`. This will be used as an identifier and must be unique in the organization. **Warning:** Changing a workspace's name changes its URL in the API and UI.
`data.attributes.description`                 | string         | (previous value) | A description for the workspace.
`data.attributes.auto-apply`                  | boolean        | (previous value) | Whether to automatically apply changes when a Terraform plan is successful, [with some exceptions](../workspaces/settings.html#auto-apply-and-manual-apply).

> `data.attributes.allow-destroy-plan`          | boolean        | (previous value)    | Whether destroy plans can be applied to the workspace.

Reviewer: ```suggestion
`data.attributes.allow-destroy-plan`          | boolean        | (previous value)    | Whether destroy plans can be queued on the workspace.
```
